### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by CindyDorantes**
## Project requirements 🎯 🎯 
- [x] Install CanCanCan in your project.
- [x] Add a `role` column to the users table. Remember to use a migration for this.
- [x] A user can delete a post if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
- [x] A user can delete a comment if it is theirs or if they have an admin role (column `role` has value `"admin"`). Use CanCanCan for this authorization.
    - For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -37,4 +37,9 @@ def show
     @user = User.find(params[:user_id])
```
Comment (MahmoudBakr23) [inline app/controllers/posts_controller.rb:39]: - [x] Please ensure using `load_and_authorize_resource` as you make the best advantage of the CanCan gem. 🙏

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
Hello @CindyDorantes  😃, 

Great work and effort you put into this project so far.

## Highlights 🚀 

There are some excellent points that you have made perfectly. Keep up! 💪
- Well done installing and using the CanCan gem
- Nice job giving the correct roles
- Descriptive PR and README file
- Correct Linters

However, there is always room for improvement.

> Keep in mind that your long hours of work are appreciated. However, the code review process is for improving the code you wrote, and not for criticizing it. 😄

 ## Required Changes ♻️
_Please check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

N/A

Cheers and Happy coding!👏👨‍💻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (CindyDorantes) [reply app/controllers/posts_controller.rb:39]: of course, thank you!

**Review by descholar-ceo — APPROVED** [review]
Hi @CindyDorantes && @Shedrack-Sunday 🤝 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
👍🏽 Implemented all of the requested changes ✔️ 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**